packages/@aws-cdk/cloudformation-include/lib/cfn-include.ts

```diff
@@ -231,7 +289,83 @@ export class CfnInclude extends core.CfnElement {
+    }
+
+    const cfnParser = new cfn_parse.CfnParser({
+      finder,
+    });
+
+    const nestedStackAttributes = templateResources[nestedStackId];
+    const nestedStackProps = cfnParser.parseValue(nestedStackAttributes.Properties);
+
+    const nestedStackScope = new core.NestedStack(this, nestedStackId, {
+      parameters: nestedStackProps.Parameters,
+      notificationArns: nestedStackProps.NotificationArns,
+      timeout: nestedStackProps.Timeout,
+    });
+
+    if (nestedStackAttributes.CreationPolicy) {
+      throw new Error('CreationPolicy is not supported by the AWS::CloudFormation::Stack resource');
+    }
+
+    if (nestedStackAttributes.UpdatePolicy) {
+      throw new Error('UpdatePolicy is not supported by the AWS::CloudFormation::Stack resource');
+    }
+
+    const cfnOptions = nestedStackScope.nestedStackResource!.cfnOptions;
+    cfnOptions.metadata = cfnParser.parseValue(nestedStackAttributes.Metadata);
+    cfnOptions.deletionPolicy = cfnParser.parseDeletionPolicy(nestedStackAttributes.DeletionPolicy);
+    cfnOptions.updateReplacePolicy = cfnParser.parseDeletionPolicy(nestedStackAttributes.UpdateReplacePolicy);
+
+    // handle DependsOn
+    nestedStackAttributes.DependsOn = nestedStackAttributes.DependsOn ?? [];
+    const dependencies: string[] = Array.isArray(nestedStackAttributes.DependsOn) ?
+      nestedStackAttributes.DependsOn : [nestedStackAttributes.DependsOn];
+    for (const dep of dependencies) {
+      const depResource = finder.findResource(dep);
+      if (!depResource) {
+        throw new Error(`nested stack '${nestedStackId}' depends on '${dep}' that doesn't exist`);
+      }
+      nestedStackScope.nestedStackResource!.node.addDependency(depResource);
+    }
+
+    // handle Condition
+    if (nestedStackAttributes.Condition) {
+      const condition = finder.findCondition(nestedStackAttributes.Condition);
+      if (!condition) {
+        throw new Error(`nested stack '${nestedStackId}' uses Condition '${nestedStackAttributes.Condition}' that doesn't exist`);
+      }
+      cfnOptions.condition = condition;
+    }
+
+    if (this.preserveLogicalIds) {
+      // override the logical ID to match the original template
+      nestedStackScope.nestedStackResource!.overrideLogicalId(nestedStackId);
+    }
+
+    // add this stack to our template's resources, so it doesn't get added again when creating resources
+    this.resources[nestedStackId] = nestedStackScope.nestedStackResource!;
+
+    const propStack = props.nestedStacks[nestedStackId];
+    const template = new CfnInclude(nestedStackScope, nestedStackId, {
+      templateFile: propStack.templateFile,
+      nestedStacks: propStack.nestedStacks,
+    });
+
+    const includeStack: IncludedNestedStack = { stack: nestedStackScope, includedTemplate: template };
+    this.nestedStacks[nestedStackId] = includeStack;
+  }
+
+  private getOrCreateResource(logicalId: string, props?: CfnIncludeProps): core.CfnResource {
```

Comment:
Thoughts on this solution?

adjusting finder's findResource to return this

`return self.getOrCreateResource(lId, props);`

instead of this

`return self.getOrCreateResource(lId);`

makes that test pass.

Reply:
It's fine, but a little weird. `getOrCreateResource()` is an instance method, and in your solution it will be called with the same `props` argument for every instance of `CfnInclude`. So passing it through arguments, instead of the field of the `CfnInclude` class, is confusing to somebody reading the code of the class, especially since `getOrCreateResource()` already reads a bunch of private fields of `CfnInclude`.